tools/querytee/response_comparator.go

```diff
@@ -495,6 +561,9 @@ func compareScalar(expectedRaw, actualRaw json.RawMessage, queryEvaluationTime t
 }
 
 func compareSamplePair(expected, actual model.SamplePair, queryEvaluationTime time.Time, opts SampleComparisonOptions) error {
+	if expected.Timestamp < opts.SkipSamplesBefore && actual.Timestamp < opts.SkipSamplesBefore {
+		return nil
+	}
```

Comment:
[nit] It took me a while to understand why we do this check before `if expected.Timestamp != actual.Timestamp {` so I think it's worth a comment.

```suggestion
	// If the timestamp is before the configured SkipSamplesBefore then we don't even check if the timestamp is correct.
	// The reason is that the SkipSamplesBefore feature may be used to compare queries hitting a different storage and one of two storages has no historical data.
	if expected.Timestamp < opts.SkipSamplesBefore && actual.Timestamp < opts.SkipSamplesBefore {
		return nil
	}
	
```